cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -158,26 +158,31 @@ func (o *Options) config(kubeAPIServerConfig *rest.Config, kubeClient *kubernete
 
 	// Initialize admission plugins
 	o.Recommended.ExtraAdmissionInitializers = func(c *genericapiserver.RecommendedConfig) ([]admission.PluginInitializer, error) {
+		protobufLoopbackConfig := *gardenerAPIServerConfig.LoopbackClientConfig
+		if protobufLoopbackConfig.ContentType == "" {
+			protobufLoopbackConfig.ContentType = runtime.ContentTypeProtobuf
```

Comment:
I don't have much details but to my knowledge the apiservers support content negotiation. Can't we just set our defaults `application/vnd.kubernetes.protobuf,application/json` and say - first try with `application/vnd.kubernetes.protobuf`, if the apiserver does not support it switch back to `application/json`? Ref https://github.com/kubernetes/kube-state-metrics/pull/264/files#r140285479

@mvladev , do you have more insights on apiserver content negotiation and how it works?

Reply:
The `Content-Type` header can't specify multiple content types according to my knowledge, as it has to specify the exact content type used in the request's body.
IIUC, the content type of the request body can't be negotiated.

The `AcceptContentType` OTOH supports content negotiation, as you correctly pointed out.
But, as it will be set to `ContentType + ", */*"` (see above), it will automatically be negotiated, if the operator didn't specify the setting explicitly. 
So I don't see the need to overwrite it. Do you agree?

Reply:
Ok, thanks for explaining.